internal/venice-common/src/main/java/com/linkedin/venice/utils/SharedObjectFactory.java

```diff
@@ -69,10 +92,15 @@ public int size() {
    * managing an object with the identifier, returns 0
    */
   public int getReferenceCount(String identifier) {
-    if (!objectMap.containsKey(identifier)) {
-      return 0;
-    } else {
-      return objectMap.get(identifier).getReferenceCount();
+    getLock(identifier).lock();
+    try {
+      if (!OBJECT_MAP.containsKey(identifier)) {
+        return 0;
+      } else {
+        return OBJECT_MAP.get(identifier).getReferenceCount();
+      }
```

Comment:
Can we replace by a `get` followed by a null check, rather than `containsKey` followed by `get`?